### PR TITLE
Doc: add python debug strategy in troubleshooting section

### PR DESCRIPTION
**Pull request by sfavazza-duagon**
As mentioned in  #1218 I add few words in the troubleshooting section to ease the debugging on the user side.

### CODE REVIEW

File: documentation/source/troubleshooting.rst [hunk 1]
```diff
@@ -15,14 +15,51 @@ Increasing Verbosity
...
+
+Python
+------
+
+When executing the Makefile to run a cocotb test, a python shell interpreter is called from within the
+VPI/VHPI/FLI library. Hence it is not possible to directly attach a python debugger to the python process being
+part of the simulator that uses the aforementioned library.
+
+To successfully debug your python code use the `remote_pdb`_ python package to create a :command:`pdb` instance
+accessible via a TCP socket:
```
Comment (eric-wieser) [inline documentation/source/troubleshooting.rst]: It's probably worth adding a comment here saying that "Using `import pdb; pdb.set_trace()` directly is also frequently not possible, due to the way that simulators interfere with stdin."

File: documentation/source/troubleshooting.rst [hunk 2]
```diff
@@ -15,14 +15,51 @@ Increasing Verbosity
 If things fail in the VPI/VHPI/FLI area, check your simulator's documentation to see if it has options to
 increase its verbosity about what may be wrong. You can then set these options on the :command:`make` command line
 as :make:var:`COMPILE_ARGS`, :make:var:`SIM_ARGS` or :make:var:`EXTRA_ARGS` (see :doc:`building` for details).
+Moreover the scheduler activity can be enhanced by setting the variable :make:var:`COCOTB_SCHEDULER_DEBUG` to
+one.
```
Comment (eric-wieser) [inline documentation/source/troubleshooting.rst]: "activity can be enhanced" is a weird way to phrase this. Perhaps
```rst
If things fail from within python, or coroutines aren't being called when you expect, the :make:var:`COCOTB_SCHEDULER_DEBUG` variable can be used to (greatly) increase the verbosity of the scheduler.
```

File: documentation/source/troubleshooting.rst [hunk 3]
```diff
@@ -15,14 +15,51 @@ Increasing Verbosity
...
 
 
 Attaching a Debugger
 ====================
 
+C-Code
```
Comment (eric-wieser) [inline documentation/source/troubleshooting.rst]:
```suggestion
C and C++
```

File: documentation/source/troubleshooting.rst (hunk 1 above)
Comment (sfavazza-duagon) [reply documentation/source/troubleshooting.rst]: I haven't had this kind of experience, I don't want to write about something I am not aware of.
After all you are a maintainer of this repository and can enrich the information wherever you consider it necessary.

File: documentation/source/troubleshooting.rst (hunk 2 above)
Comment (sfavazza-duagon) [reply documentation/source/troubleshooting.rst]: Yeah I forgot to double check. Sure it does not make sense as I wrote it.
I will commit the suggestion.

File: documentation/source/troubleshooting.rst (hunk 1 above)
Comment (eric-wieser) [reply documentation/source/troubleshooting.rst]: I thought you said you'd tried `pdb` with no success.

Comment (sfavazza-duagon) [reply documentation/source/troubleshooting.rst]: Honestly I did not really get what is going under the hood, so I don't find comfortable to describe something I do not understand fully, moreover the remote debugging it should always work as it create a socket at system level, independently of the simulator it is used.
I described only the successful recipe.
I hope you are good with that.

Comment (sfavazza-duagon) [reply documentation/source/troubleshooting.rst]: If it is so important to add such information, could suggest a sentence to add to my pull request?

**Comment by sfavazza-duagon** [comment]
I think that many people would profit from the debug strategy section. Can you kindly merge it?

**Comment by cmarqu** [comment]
I'll do so later today/tonight.

File: documentation/source/troubleshooting.rst [hunk 4]
```diff
@@ -15,14 +15,51 @@ Increasing Verbosity
 If things fail in the VPI/VHPI/FLI area, check your simulator's documentation to see if it has options to
 increase its verbosity about what may be wrong. You can then set these options on the :command:`make` command line
 as :make:var:`COMPILE_ARGS`, :make:var:`SIM_ARGS` or :make:var:`EXTRA_ARGS` (see :doc:`building` for details).
+If things fail from within python, or coroutines aren't being called when you expect, the
```
Comment (cmarqu) [inline documentation/source/troubleshooting.rst]:
```suggestion
If things fail from within Python, or coroutines aren't being called when you expect, the
```

File: documentation/source/troubleshooting.rst [hunk 5]
```diff
@@ -15,14 +15,52 @@ Increasing Verbosity
...
 you can set the environment variable :envvar:`COCOTB_ATTACH` to a pause time value in seconds.
 If set, cocotb will print the process ID (PID) to attach to and wait the specified time before
 actually letting the simulator run.
 
 For the GNU debugger GDB, the command is :command:`attach <process-id>`.
+
+Python
+------
+
+When executing the Makefile to run a cocotb test, a python shell interpreter is called from within the
```
Comment (cmarqu) [inline documentation/source/troubleshooting.rst]:
```suggestion
When executing the Makefile to run a cocotb test, a Python shell interpreter is called from within the
```

**Comment by cmarqu** [comment]
Thanks very much for the contribution, and sorry for the delay in merging.

(Also, I now know Github's web GUI a bit better.)

**Pull request merged**